src/cargo/ops/cargo_rustc/custom_build.rs

```diff
@@ -91,7 +91,8 @@ pub fn prepare(pkg: &Package, target: &Target, req: Platform,
         let not_custom = pkg.targets().iter().find(|t| {
             !t.is_custom_build()
         }).unwrap();
-        cx.dep_targets(pkg, not_custom, profile).iter().filter_map(|&(pkg, t, _)| {
+        cx.dep_targets(pkg, not_custom, Kind::Host, profile)
```

Comment:
I think this wants to actually be `kind` (the local variable above) instead of `Kind::Host` because these are the inputs to the build script which are those deps which are linked for the architecture the build script is targeting.

To help catch this, can you write a test that looks like:

* Package A has a build script and platform-specific dependencies B/C (both for different platforms)
* Packages B and C both have build scripts with some variables
* When package A is compiled for B's architecture, it shouldn't see C's variables but should see B's variables
* When package A is compiled for C's architecture, it shouldn't see B's variables but should see C's variables